graph/multi/multi.go

```diff
@@ -0,0 +1,112 @@
+// Copyright ©2014 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multi
+
+import "gonum.org/v1/gonum/graph"
+
+// Node here is a duplication of simple.Node
+// to avoid needing to import both packages.
+
+// Node is a simple graph node.
+type Node int64
+
+// ID returns the ID number of the node.
+func (n Node) ID() int64 {
+	return int64(n)
+}
+
+// Edge is a collection of multigraph edges sharing end points.
+type Edge []graph.Line
+
+// From returns the from-node of the edge.
```

Comment:
Add a comment along the lines of "If more than one edge exist, the from node of the first edge is returned." Likewise for `To`.

Reply:
This is an implementation detail that I would rather not document; if people are doing things that make that behaviour distinguishable, they are holding it wrong. This does suggest to me though a way that we can deal with the node edge ordering subtlety. I am thinking that perhaps graph.Edge should be able to set nodes irrespective of whether it is a constructor type. This would mean that a properly written edge type would always have the correct node behaviour. I'll send comments to that issue later.

Reply:
Sounds good. Thanks for the explanation.